idaes/apps/uncertainty_propagation/uncertainties.py

```diff
@@ -0,0 +1,371 @@
+    # get the column numbers of theta
+    line_dic = {}
+    for v in theta_names:
+        line_dic[v] = line_num('col_row.col', v)
+    # load gradient of the objective function
+    gradient_f = np.loadtxt("./GJH/gradient_f_print.txt")
+    # load gradient of all constraints (sparse)
+    # If no constraint exists, return []
+    num_constraints = len(list(model.component_data_objects(Constraint,
+                                                            active=True,
+                                                            descend_into=True)))
+    if num_constraints > 0 :
+        gradient_c = np.loadtxt("./GJH/A_print.txt")
+    else:
+        gradient_c = np.array([])
+
+    # remove all generated files
+    shutil.move("col_row.nl", "./GJH/")
+    shutil.move("col_row.col", "./GJH/")
+    shutil.move("col_row.row", "./GJH/")
+    shutil.rmtree('GJH', ignore_errors=True)
+    return gradient_f, gradient_c, line_dic
+
+def line_num(file_name, target):
+    """This function returns the line number contains 'target' in the file_name
+    Parameters
+    ----------
+    file_name: string
+        file name includes information of variabe order (col_row.col)
+    target: string   
+        variable name to check  
+    Returns
+    -------
+    count: int
+        line number of target in the file_name
+        
+    Raises
+    ------
+    Exception
+        When col_row.col doesnot include target
+    """
+    with open(file_name) as f:
+        count = int(1)
+        for line in f:
+            if line.strip() == target:
+                return int(count)
+            count += 1
+    raise Exception("col_row.col should includes target")
+
+def clean_variable_name(theta_names):
+    """This eunction removes all ' and spaces in theta_names.
+       Note: The  current theta_est(calc_cov=True) of parmest in Pyomo
+       doesn't allow ' and spaces in the variable names    
+    Parameters
+    ----------
+    theta_names: list of strings
+        List of Var names
+    
+    Returns
+    -------
+    theta_names_out: list of strings
+        List of Var names after removing  all ' and spaces
+    var_dic: dict
+       dictionary with keys converted theta_names and values origianl theta_names 
+    """
+
+    # Variable names cannot have "'" for parmest_class.theta_est(calc_cov=True)
+    # Save original variables name in to var_dic
+    # Remove all "'" and " " in theta_names
+    var_dic = {}
+    theta_names_out = []
+    clean = False
+    for i in range(len(theta_names)):
+        if "'" in theta_names[i] or " " in theta_names[i] :
+            logger.warning(theta_names[i] + " includes ' or space.")
+            clean = True
+        theta_tmp = theta_names[i].replace("'", '')
+        theta_tmp = theta_tmp.replace(" ", '')
+        theta_names_out.append(theta_tmp)
+        var_dic[theta_tmp] = theta_names[i]
+    if clean:
+       logger.warning("All ' and spaces in theta_names are removed.")
```

Comment:
Add more information including the original and cleaned names.

Reply:
Actually, I see you are already giving the original variable name. I recommend giving the new one too.

Reply:
I added a logger to provide the original variable name and the cleaned variable name if clean is True.